File: lib/src/arcore_sessionconfig.dart
class ARCoreSessionConfig {
  final PlaneFindingMode planeFindingMode;

  const ARCoreSessionConfig({
    this.planeFindingMode,
  });

  Map<String, dynamic> toMap() => <String, dynamic>{
        'planeFindingMode': planeFindingMode._value,
      };
}

class PlaneFindingMode {
  static const PlaneFindingMode DISABLED = PlaneFindingMode._(0, 'DISABLED');
  static const PlaneFindingMode HORIZONTAL =
      PlaneFindingMode._(1, 'HORIZONTAL');
  static const PlaneFindingMode VERTICAL = PlaneFindingMode._(2, 'VERTICAL');
  static const PlaneFindingMode HORIZONTAL_AND_VERTICAL =
      PlaneFindingMode._(3, 'HORIZONTAL_AND_VERTICAL');
  static final values = [
    DISABLED,
    HORIZONTAL,
    VERTICAL,
    HORIZONTAL_AND_VERTICAL,
  ];

  const PlaneFindingMode._(this._value, this._text);

  final int _value;
  final String _text;

  static PlaneFindingMode get(int value) {
    PlaneFindingMode ret = DISABLED;
    values.forEach((m) {
      if (m._value == value) {
        ret = m;
        return;
      }
    });
    return ret;
  }

  @override
  String toString() {
    return '$_text - $_value';
  }
}

File: android/src/main/kotlin/com/difrancescogianmarco/arcore_flutter_plugin/ARCoreSessionConfig.kt
package com.difrancescogianmarco.arcore_flutter_plugin

import com.google.ar.core.Config

class ARCoreSessionConfig private constructor(val planeFindingMode: Config.PlaneFindingMode) {

    companion object {
        fun fromMap(map: Map<*, *>): ARCoreSessionConfig {
            val planeNum = (map["planeFindingMode"] as? Int) ?: 0
            val plane = when (planeNum) {
                1 -> Config.PlaneFindingMode.HORIZONTAL
                2 -> Config.PlaneFindingMode.VERTICAL
                3 -> Config.PlaneFindingMode.HORIZONTAL_AND_VERTICAL
                else -> Config.PlaneFindingMode.DISABLED
            }
            return ARCoreSessionConfig(plane)
        }

        val defaultConfig = ARCoreSessionConfig(Config.PlaneFindingMode.DISABLED)
    }

    override fun toString(): String {
        return "planeFindingMode: $planeFindingMode"
    }
}